src/use_case_classification.Rmd

```diff
@@ -0,0 +1,147 @@
+
+#### Preprocessing
+
+The data set is corrected regarding their data types.
+
+```{r}
+data[, c("Survived", "Pclass", "Sex", "SibSp", "Embarked")] = lapply(data[,  c("Survived", "Pclass", "Sex", "SibSp", "Embarked")], as.factor)
+```
+
+Next, unuseful columns will be dropped.
+
+```{r}
+data = dropNamed(data, c("Cabin","PassengerId", "Ticket", "Name"))
+```
+
+And missing values will be imputed, in this case Age and Fare.
+
+```{r}
+data$Embarked[data$Embarked == ""] = NA
+data$Embarked = droplevels(data$Embarked)
+data = impute(data, cols = list(Age = imputeMedian(), Fare = imputeMedian(), Embarked = imputeMode()))
+data = data$data
+```
+
+### Create a task
+
+In the "task" the data set and the target column is specified. People who survived are labelled with "1".
+
+```{r}
+task = makeClassifTask(data = data, target = "Survived", positive = "1")
+```
+
+### Define a learner
+
+A classification learner is selected. You can find an overview of all learners [here](http://mlr-org.github.io/mlr-tutorial/devel/html/integrated_learners/index.html)
+
+```{r}
+lrn = makeLearner("classif.randomForest", predict.type = "prob")
+```
+
+### Fit the model
+
+To fit the model - and afterwards predict - the data set is split into a training and a test data set.
+
+```{r}
+n = getTaskSize(task)
+trainSet = seq(1, n, by = 2)
+testSet = seq(2, n, by = 2)
+```
+
+```{r}
+mod = train(learner = lrn, task = task, subset = trainSet)
+```
+
+### Predict
+
+Predicting the target values for new observations is implemented the same way as most of the other predict methods in R. In general, all you need to do is call predict on the object returned by train and pass the data you want predictions for.
+
+```{r}
+pred = predict(mod, task, subset = testSet)
+```
+
+The quality of the predictions of a model in mlr can be assessed with respect to a number of different performance measures. In order to calculate the performance measures, call performance on the object returned by predict and specify the desired performance measures.
+
+```{r}
+calculateConfusionMatrix(pred)
+performance(pred, measures = list(acc, fpr, tpr))
+df = generateThreshVsPerfData(pred, list(fpr, tpr, acc))
+plotThreshVsPerf(df)
+plotROCCurves(df)
+```
+
+### Extension of the original use case
+
+As you might have seen the titanic library also provides a second dataset.
+
+```{r}
+test = titanic_test
+head(test)
+```
+
+This one does not contain any survival information, but we now can use our fitted model and predict the survival probability for this data set.
+
+The same preprocessing steps - as for the "data" data set - have to be applied
+
+```{r}
+test[, c("Pclass", "Sex", "SibSp", "Embarked")] = lapply(test[,  c("Pclass", "Sex", "SibSp", "Embarked")], as.factor)
+
+test = dropNamed(test, c("Cabin","PassengerId", "Ticket", "Name"))
+
+test = impute(test, cols = list(Age = imputeMedian(), Fare = imputeMedian()))
```

Comment:
I'm not totally happy about how you are doing the imputation.
1. For titanic_train you are imputing on the **whole** data set **before** splitting it into train and test samples.
2. You are doing the imputation **separately** for titanic_train and titanic_test.

I think the cleanest way would be imputation on the train samples and reimputation (using the estimated means and medians from the train samples) on the test samples, respectively. This is easy to do with an imputation wrapper, but  the less beginner-friendly variant.

Therefore, I would stick with approach 2., but use it consistently.